### ISSUE TITLE
CSS: Comment lines affect selector nesting

### ISSUE THREAD

**Issue by infotexture** [prettier/prettier#3388]
### 1. Long comment lines nest selectors

The placement (and length) of comment lines trips up the selector nesting such that a second selector is indented like a descendant class.

**Prettier 1.8.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVAAgGYEsBOAzjJpALZkInqoA6UAxHkTALSFwA2cYME+AGnoZMHSFAAmpCBSqYYACwCGMAOSFMnaAHNMCCAFdtC+REwAHA4ROK4ontCkduvfqcxQ4Ad025PmGnoAOjFHdi4ePnxMYHpMTAlcQnNOJQBPJA9oOwBCXDJzfhglWABuegBfehABEAhzGFxoQmRQJWJkbCVODlqAI3wlMABrOBgAZXMhv21kGHwDOFqJCDBO7t6QPw58GAAFQe0yJXWepZAAK0IADwAhQZGx8aVKABk-OFPNwxhLGAAmL7nKZEOD4ZDgQgtWrmfB+GAAdVwEkUyAAHAAGGH4CAcBGDcxzBbnfBwACOBgIcAOSiOJyQXTOtQ4ZFwRMWzJm3AAigYIPAgbVin0kSiFMh-kLBrhODMAMIyY4QqDZGogKxwAAqSj6LQZGzgFQqQA)
```sh
--parser css
```

**Input:**
```css
/* first comment */
#first-selector,
/* second comment that's long enough to push the second selector to new line */
.second-selector {
  display: none !important;
}

```

**Output:**
_(second selector is incorrectly indented like a descendant class)_
```css
/* first comment */
#first-selector,
/* second comment that's long enough to push the second selector to new line */
  .second-selector {
  display: none !important;
}

```

**Expected behavior (input should remain unchanged):**
```css
/* first comment */
#first-selector,
/* second comment that's long enough to push the second selector to new line */
.second-selector {
  display: none !important;
}

```

If the second comment is moved down one line, the parser correctly recognizes the second selector and does not indent it:

```css
/* first comment */
#first-selector,
.second-selector {
  /* second comment that's long enough to push the second selector to new line */
  display: none !important;
}

```

### 2. Short comments & selectors are combined

If the second comment (and the following selector) are short enough, they appear on the same line after Prettier runs:

**Input:**
```css
/* first comment */
#first-selector,
/* second comment that's short */
.second-selector {
  display: none !important;
}
```

**Output:**
_(second comment & second selector are concatenated to a single line)_

```css
/* first comment */
#first-selector,
/* second comment that's short */ .second-selector {
  display: none !important;
}
```

This adversely affects readability of the code, since the comment makes it difficult to recognize the second selector.

Comments that appear on a single line in input should be output on a line of their own (regardless of their length), not combined with selectors.

**Comment by lydell** [comment]
FYI the comments support is pretty bad overall. For now you’re better off sticking to “simpler” comment locations: between rules and between declarations.

**Comment by infotexture** [comment]
Yeah, I realize this is probably an edge case, but figured it was worth mentioning nonetheless.

**Comment by evilebottnawi** [comment]
Should be fixed first: https://github.com/postcss/postcss-scss/issues/75

**Comment by glen-84** [comment]
This doesn't seem to be an issue anymore.

**Comment by thorn0** [comment]
Fixed long time ago (tested it with Prettier 1.15).
